dl4j-examples/src/main/java/org/deeplearning4j/examples/nlp/sentencepiecernnexample/SentencePieceRNNExample.java

```diff
@@ -0,0 +1,260 @@
+//           // two Bidirectional LSTM layers in a row with dropout and tanh as activation function
+            .layer(new Bidirectional(new LSTM.Builder().nOut(256)
+                .dropOut(0.8).activation(Activation.TANH).build()))
+            .layer(new Bidirectional(new LSTM.Builder().nOut(256)
+                .dropOut(0.8).activation(Activation.TANH).build()))
+            .layer(new GlobalPoolingLayer(PoolingType.MAX))
+            // defining last layer with 2 outputs (2 classes - positive and negative),
+            // small dropout to avoid overfitting and MCXENT loss function
+            .layer(new OutputLayer.Builder().nOut(2)
+                .dropOut(0.97).activation(Activation.SOFTMAX)
+                .lossFunction(LossFunctions.LossFunction.MCXENT).build())
+            .build();
+
+
+        // Getting train and test BertIterators for both: test and train,
+        // changing argument isTraining: true to get train and false to get test respectively
+        BertIterator train = getBertDataSetIterator(true, t);
+        BertIterator test = getBertDataSetIterator(false, t);
+
+
+        // preprocessor for DataType matching
+        MultiDataSetPreProcessor mdsPreprocessor = new MultiDataSetPreProcessor() {
+            @Override
+            public void preProcess(MultiDataSet multiDataSet) {
+                multiDataSet.setFeaturesMaskArray(0, multiDataSet.getFeaturesMaskArray(0).castTo(DataType.FLOAT));
+            }
+        };
+
+        // Applying preprocessor for both: train and test datasets
+        train.setPreProcessor(mdsPreprocessor);
+        test.setPreProcessor(mdsPreprocessor);
+
+        // initialize MultiLayerNetwork instance with described above configuration
+        MultiLayerNetwork net = new MultiLayerNetwork(conf);
+
+
+        //Initialize the user interface backend
+        UIServer uiServer = UIServer.getInstance();
+
+        //Configure where the network information (gradients, activations, score vs. time etc) is to be stored
+        //Then add the StatsListener to collect this information from the network, as it trains
+        StatsStorage statsStorage = new FileStatsStorage(new File(System.getProperty("java.io.tmpdir"), "ui-stats-" + System.currentTimeMillis() + ".dl4j"));
+        int listenerFrequency = 20;
+        net.setListeners(new StatsListener(statsStorage, listenerFrequency), new ScoreIterationListener(50));
+        //Attach the StatsStorage instance to the UI: this allows the contents of the StatsStorage to be visualized
+        uiServer.attach(statsStorage);
+
+
+        // Setting to train net for 19 epochs (note: previous net state persist after each iteration)
+        for (int i = 1; i <= 19; i++) {
+
+            net.fit(train);
+
+            // Get and print accuracy, precision, recall & F1 and confusion matrix
+            Evaluation eval = net.doEvaluation(test, new Evaluation[]{new Evaluation()})[0];
+            System.out.println(eval.stats());
```

Comment:
This might be a slight improvement?
```
System.out.println("===== Evaluation at training iteration " + i + " =====");
System.out.println(eval.stats());
```